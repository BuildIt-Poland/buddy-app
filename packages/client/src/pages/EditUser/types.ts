import { RouteComponentProps } from 'react-router-dom';

export type EditUserProps = Pick<RouteComponentProps, 'history'>;
