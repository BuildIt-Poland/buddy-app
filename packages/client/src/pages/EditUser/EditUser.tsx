import React from 'react';
import { useParams } from 'react-router-dom';
import { useQuery, useMutation } from '@apollo/react-hooks';
import {
  Mutation,
  QueryBuddyArgs,
  QueryNewbieArgs,
  UserInput,
} from '@buddy-app/schema';
import { UPDATE_USER_MUTATION } from 'graphql/update-user.graphql';
import { useSnackBar } from 'contexts/SnackbarContext';
import UserForm from 'components/UserForm';
import FormPlaceHolder from 'atoms/FormPlaceHolder';
import BackPageContainer from 'atoms/BackPageContainer';
import { useAuth } from 'contexts/AuthContext';
import { goBack } from 'utils';
import {
  getQueryData,
  BasicDetailsParams,
  UserBasicDetails,
} from 'pages/ContactDetails';
import DICTIONARY from './dictionary';
import { EditUserProps } from './types';

const EditUser: React.FC<EditUserProps> = ({ history }) => {
  const {
    data: { role, userId },
  } = useAuth();
  const { newbieId, buddyId } = useParams<QueryNewbieArgs & QueryBuddyArgs>();
  const { showSnackbar } = useSnackBar();

  const { query, variables, userRole } = getQueryData(userId, newbieId, buddyId)[
    role
  ];

  const { data, loading } = useQuery<UserBasicDetails, BasicDetailsParams>(query, {
    variables,
  });

  const userDetails = data && data[userRole.toLowerCase()];

  const [updateUser, { loading: updateLoading }] = useMutation<Partial<Mutation>>(
    UPDATE_USER_MUTATION,
    {
      onCompleted: () => {
        showSnackbar(DICTIONARY.DIALOG.SUCCESS_MSG);
        goBack(history);
      },
      onError: error => {
        showSnackbar((error && error.message) || DICTIONARY.DIALOG.ERROR_MSG);
      },
    }
  );

  const onSubmit = (input: UserInput) => {
    updateUser({
      variables: {
        userId: newbieId || buddyId || userId,
        input,
      },
    });
  };

  return (
    <BackPageContainer title={DICTIONARY.TITLE} id='add-user-page' backGroundShape>
      {loading && <FormPlaceHolder />}
      {userDetails && (
        <UserForm
          loading={updateLoading}
          isNewbie={!!newbieId}
          data={userDetails}
          onSubmit={onSubmit}
        />
      )}
    </BackPageContainer>
  );
};

export default EditUser;
