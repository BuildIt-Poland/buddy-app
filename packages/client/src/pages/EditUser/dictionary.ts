interface Dictionary {
  TITLE: string;
  DIALOG: {
    SUCCESS_MSG: string;
    ERROR_MSG: string;
  };
}

const DICTIONARY: Dictionary = {
  TITLE: 'Update Contact Details',
  DIALOG: {
    SUCCESS_MSG: 'Contact details have been successfully updated',
    ERROR_MSG: 'An error ocurred updating contact details',
  },
};

export default DICTIONARY;
