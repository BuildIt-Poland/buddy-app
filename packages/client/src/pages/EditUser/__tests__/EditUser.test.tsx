import React from 'react';
import { create } from 'react-test-renderer';

import EditUser from '../EditUser';

describe('Component - EditUser', () => {
  test('renders correctly', () => {
    const component = create(<EditUser />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
