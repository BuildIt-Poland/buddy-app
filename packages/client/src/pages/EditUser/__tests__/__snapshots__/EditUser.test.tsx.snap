// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - AddUser renders correctly 1`] = `
<div
  data-testid="add-user-page"
>
  <h1>
    Add a User
  </h1>
</div>
`;
